src/ParseError.js

```diff
@@ -3,25 +3,43 @@
  * functions when something has gone wrong. This is used to distinguish internal
  * errors from errors in the expression that the user provided.
  */
-function ParseError(message, lexer, position) {
+function ParseError(message, token) {
     var error = "KaTeX parse error: " + message;
+    var start;
+    var end;
 
-    if (lexer !== undefined && position !== undefined) {
+    if (token && token.lexer) {
```

Comment:
I think this is fine, it's probably worth add a comment to the Token constructor which points out that `lexer` is optional and why that's desirable.